refactor: replace deprecated template provider usage

rosstimson opened the pull request:
In Terraform v0.12 and above the `terraform_file` resource that is part
of the template provider has been replaced with a built-in function
`templatefile` that is simpler to use.  You can see this mentioned in
the docs at:
https://registry.terraform.io/providers/hashicorp/template/latest/docs/data-sources/file

You can also see that the `template` provider itself has been
archived: https://github.com/hashicorp/terraform-provider-template
There will therefore be no further updates or builds for it which
becomes a problem for Mac users on the new Apple Silicon chips, it
means that wherever the old and deprecated `template` provider is used
anyone on an Apple machine with M1 chip will not be able to
successfully run `terraform init`.

I'm not sure if there is a better way to iterate over all the
variables that need to be passed into the `templatefile` function as a
map.

rosstimson committed:
refactor: replace deprecated template provider usage

In Terraform v0.12 and above the `terraform_file` resource that is part
of the template provider has been replaced with a built-in function
`templatefile` that is simpler to use.  You can see this mentioned in
the docs at:
https://registry.terraform.io/providers/hashicorp/template/latest/docs/data-sources/file

You can also see that the `template` provider itself has been
archived: https://github.com/hashicorp/terraform-provider-template
There will therefore be no further updates or builds for it which
becomes a problem for Mac users on the new Apple Silicon chips, it
means that wherever the old and deprecated `template` provider is used
anyone on an Apple machine with M1 chip will not be able to
successfully run `terraform init`.

I'm not sure if there is a better way to iterate over all the
variables that need to be passed into the `templatefile` function as a
map.

rosstimson committed:
chore: remove now unnecessary template provider in examples

nimrodkor commented:
LGTM, @JamesWoolfenden PTAL

nimrodkor commented on bridgecrew_sns_resource.tf:
```diff
@@ -1,20 +1,17 @@
 resource "null_resource" "create_bridgecrew" {
   provisioner "local-exec" {
-    command     = "aws sns ${local.profile_str} --region ${data.aws_region.region.id} publish --target-arn \"${local.bridgecrew_sns_topic}\" --message '${jsonencode(data.template_file.message.rendered)}' && sleep 30"
+    command     = "aws sns ${local.profile_str} --region ${data.aws_region.region.id} publish --target-arn \"${local.bridgecrew_sns_topic}\" --message '${jsonencode(templatefile("${path.module}/message.json.tpl", { request_type = local.request_type, bridgecrew_sns_topic = local.bridgecrew_sns_topic, customer_name = local.customer_name, account_id = local.account_id, external_id = local.external_i ...
```
@rosstimson Can we save the result of `templatefile()` as a local var, and then reuse it across this file?

rosstimson replied:
I hope https://github.com/bridgecrewio/terraform-aws-bridgecrew-read-only/pull/8/commits/16a53005aa4c28d4a1c45ada24fe5abb76e540dd is what you were after there.  It seems to work for me but will need some further testing.

nimrodkor reviewed (commented):
Looks really good, tnx @rosstimson !
Just 1 comment 🙏

rosstimson committed:
refactor: store templatefile function result as a local var

As requested by @nimrodkor store the result of the templatefile()
function in a local variable and use that throughout instead of so
much repetition. This makes the templatefile function call itself more
readable too.

JamesWoolfenden reviewed (approved):
LGTM
